src/BenchmarkDotNet/Templates/WasmBenchmarkProgram.txt

```diff
@@ -0,0 +1,121 @@
+$ShadowCopyDefines$
+$ExtraDefines$
+// <auto-generated />
+$AdditionalLogic$
+// this file must not be importing any namespaces
+// we should use full names everywhere to avoid any potential naming conflicts, example: #1007, #778
+
+// the namespace name must be in sync with WindowsDisassembler.BuildArguments
+namespace BenchmarkDotNet.Autogenerated
+{
+    public class UniqueProgramName // we need different name than typical "Program" to avoid problems with referencing "Program" types from benchmarked code, #691
+    {
+        $ExtraAttribute$
+        public static System.Int32 Main(System.String[] args)
+        {
+            // this method MUST NOT have any dependencies to BenchmarkDotNet and any other external dlls! (CoreRT is exception from this rule)
+            // otherwise if LINQPad's shadow copy is enabled, we will not register for AssemblyLoading event
+            // before .NET Framework tries to load it for this method
+#if NETFRAMEWORK
+            using(new DirtyAssemblyResolveHelper())
+#endif
+                return AfterAssemblyLoadingAttached(args);
+        }
+
+        private static System.Int32 AfterAssemblyLoadingAttached(System.String[] args)
+        {
+            BenchmarkDotNet.Engines.ConsoleHost host = new BenchmarkDotNet.Engines.ConsoleHost(System.Console.Out, System.IO.TextReader.Null); // this variable name is used by CodeGenerator.GetCoreRtSwitch, do NOT change it
```

Comment:
Input is used for acknowledgements.

Reply:
We will need to use a "FakeHost" instead of console host, use it with an ifdef to crate a new one of those insteaad of ConsoleHost

Reply:
Introduce a WasmRuntime type to propagate ifdefs; CodeGenerator.cs will have to check this type.